### ISSUE TITLE
Tests fail when branch is master

### ISSUE THREAD

**Issue by ataylorme** [pantheon-systems/terminus-build-tools-plugin#72]
@stevector can you explain what's going on in [lines 7-12 of `.circleci/test-example-repo.sh`](https://github.com/pantheon-systems/terminus-build-tools-plugin/blob/master/.circleci/test-example-repo.sh#L7:L12)?

`grep -v '^\(master\|[0-9]\+.x\)$'` is an inverted match so the `master` branch is being ignored. Also `BRANCH` is used for `PR_BRANCH` and ultimately `BUILD_TOOLS_VERSION`. Seems to me this was meant to detect the `1.x` vs. `2.x` versions. Perhaps if the branch is `master` we can have the version be `2`, otherwise fallback to the logic to check the version.

cc @greg-1-anderson

**Comment by ataylorme** [comment]
See [build 348](https://circleci.com/gh/pantheon-systems/terminus-build-tools-plugin/348) for an example of the failure.

**Comment by greg-1-anderson** [comment]
This script is running with `#!/bin/bash -eo pipefail`, which is a good practice. However, it does mean that any command that returns with a non-zero exit code will cause the entire build to fail.

This script inherited some techniques from older scripts that use commands such as grep as a filter; the previous script did not run with `-eo pipefail`, and the grep command sometimes returns with a non-zero exit code, and this was not intended to cause the script to fail.

We could either delay setting `-e` until after the variables are set up, or use some technique other than `grep` to differentiate between PR and main branch builds.

**Comment by ataylorme** [comment]
> or use some technique other than grep to differentiate between PR and main branch builds.

We should use [CircleCI environment variables](https://circleci.com/docs/2.0/env-vars/#build-details). We can just check if `CIRCLE_PULL_REQUEST` is populated.

**Comment by ataylorme** [comment]
@greg-1-anderson what is the intention with

```
BRANCH=$(echo $CIRCLE_BRANCH | grep -v '^\(master\|[0-9]\+.x\)$')
PR_BRANCH=${BRANCH:+dev-$BRANCH}
BUILD_TOOLS_VERSION=${PR_BRANCH:-$CIRCLE_BRANCH}
```

I don't see `BUILD_TOOLS_VERSION` used anywhere and am not clear what `PR_BRANCH` should be for `master` vs. another branch

**Comment by greg-1-anderson** [comment]
`BUILD_TOOLS_VERSION` is used to ensure that the version of the Terminus Build Tools plugin used in the tests for the system-under-test (the GitHub repo + Pantheon site + Circle configuration) is the same as the version being tested by the PR.

See: https://github.com/pantheon-systems/terminus-build-tools-plugin/blob/master/.circleci/test-example-repo.sh#L15

These variables were calculated via `grep` in Circle 1.x because they had to be done on a single line, when the variables were evaluated. Since we are using a script in Circle 2.x, we could simply use `if` statements instead.

**Comment by greg-1-anderson** [comment]
Fixed by #73
